sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c

```diff
@@ -52,14 +64,128 @@
  * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
  */
 
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/netif.h"
+#include "net/gnrc/netif/hdr.h"
+#include "net/gnrc/udp.h"
 #include "net/gnrc/ipv6/nib.h"
 #include "net/gnrc/ndp.h"
+#include "random.h"
+#include "xtimer.h"
+
+/**
+ * @brief Port for the custom UDP sync protocol
+ */
+#ifndef CONFIG_GNRC_IPV6_AUTO_SUBNETS_PORT
+#define CONFIG_GNRC_IPV6_AUTO_SUBNETS_PORT          (16179)
+#endif
+
+/**
+ * @brief Max number of other routers on the same link
+ */
+#ifndef CONFIG_GNRC_IPV6_AUTO_SUBNETS_PEERS_MAX
+#define CONFIG_GNRC_IPV6_AUTO_SUBNETS_PEERS_MAX     (4)
+#endif
+
+/**
+ * @brief How often the number subnets should be announced by the routers
+ */
+#ifndef CONFIG_GNRC_IPV6_AUTO_SUBNETS_TX_RETRIES
+#define CONFIG_GNRC_IPV6_AUTO_SUBNETS_TX_RETRIES    (3)
+#endif
+/**
+ * @brief How long to wait for other routers annoucements before resending
+ *        or creating subnets when the retry counter is exhausted
+ */
+#ifndef CONFIG_GNRC_IPV6_AUTO_SUBNETS_TIMEOUT_MS
+#define CONFIG_GNRC_IPV6_AUTO_SUBNETS_TIMEOUT_MS    (50)
+#endif
+
+#define SERVER_THREAD_STACKSIZE                     (THREAD_STACKSIZE_DEFAULT)
+#define SERVER_MSG_QUEUE_SIZE                       (CONFIG_GNRC_IPV6_AUTO_SUBNETS_PEERS_MAX)
+#define SERVER_MSG_TYPE_TIMEOUT                     (0x8fae)
 
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
 static char addr_str[IPV6_ADDR_MAX_STR_LEN];
 
+#if !IS_USED(MODULE_GNRC_IPV6_AUTO_SUBNETS_SIMPLE)
+
+/**
+ * @brief Custom UDP sync protocol
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t num_subnets;    /**< number of subnets a host wants to create */
+} auto_subnets_request_t;
```

Comment:
> Is a type for a single byte header really necessary?

I thought it would be cleaner that way in case the format needs to be extended in the future.

Reply:
> […] in case the format needs to be extended in the future.

Maybe add a version field in that case. You will thank me later, if this is deployed and not all nodes were updated yet ;-).

Reply:
I think the upper nibble of the first byte is enough for that (like in IP). The remaining bits can be reserved for future use as well.